coderadar-graph/src/main/java/io/reflectoring/coderadar/graph/contributor/repository/ContributorRepository.java

```diff
@@ -0,0 +1,27 @@
+package io.reflectoring.coderadar.graph.contributor.repository;
+
+import io.reflectoring.coderadar.graph.contributor.domain.ContributorEntity;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.neo4j.annotation.Query;
+import org.springframework.data.neo4j.repository.Neo4jRepository;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ContributorRepository extends Neo4jRepository<ContributorEntity, Long> {
+  @Query("MATCH (c) WHERE ID(c) = {0} RETURN c")
+  @NonNull
+  Optional<ContributorEntity> findById(@NonNull Long id);
+
+  @Query("MATCH (c:ContributorEntity)-[:WORKS_ON]->(p:ProjectEntity) WHERE ID(p) = {0} RETURN c")
+  List<ContributorEntity> findAllByProjectId(Long projectId);
+
+  @Query(
+      "MATCH (c:ContributorEntity)-[:WORKS_ON]->(p:ProjectEntity)-[:CONTAINS_COMMIT]->(co:CommitEntity)<-[:CHANGED_IN]-(f:FileEntity)"
+          + " WHERE ID(p) = {0} AND f.path ENDS WITH {1} AND co.author IN c.names RETURN c")
```

Comment:
I think none of the node labels here are nessecery (you should profile this however). Also I'm not 100% sure about the way you check for Filenames with `ENDS WITH`. Do we care if two files have the same name? If not then this is correct. Also, do we care if a file has been renamed?